### PR TITLE
Templates and single binary

### PR DESCRIPTION
**Pull request by calj**
- Template the project with project name, organization and git repository domain
- Create a single binary with server and cli commands

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -28,17 +42,37 @@ module.exports = class extends Generator {
...
   }
 
-  install() {
-    this.installDependencies();
+  end() {
+    this.spawnCommandSync("git", ["init", "--quiet"], {
+      cwd: this.destinationPath()
+    });
+
+    this.spawnCommandSync("git", ["add", "."], {
```
Comment (mod) [inline generators/app/index.js:90]: I suggest make this a function and we make git optional
also better we make sure it builds before commiting ?

File: generators/app/templates/cmd/sonic/sonic.go [hunk 2]
```diff
@@ -4,7 +4,10 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/openware/igonic/config"
+  "github.com/foolin/goview/supports/ginview"
+	"github.com/gin-gonic/gin"
+	"<%= git_domain %>/<%= organization %>/<%= project %>/config"
+	"<%= git_domain %>/<%= organization %>/<%= project %>/routes"
```
Comment (mod) [inline generators/app/templates/cmd/sonic/sonic.go]: I suggest making a function or simply a prop : <%= package_name %>
project name being the last element of the url

**Review by mod — CHANGES_REQUESTED** [review]
very interesting start, we can merge but I feel it's good to comment more here and collaborate

File: generators/app/index.js (hunk 1 above)
Comment (calj) [reply generators/app/index.js:90]: Added --nogit argument
Then I think we can keep it simple, if the result is not perfect the operator will fix and eventually PR to the generator.

File: generators/app/templates/cmd/sonic/sonic.go (hunk 2 above)
Comment (calj) [reply generators/app/templates/cmd/sonic/sonic.go]: done

**Review by mod — APPROVED** [review]

**Pull request merged**